lib/remap-lang/src/expression/argument.rs

```diff
@@ -0,0 +1,88 @@
+use crate::{
+    expression, expression::Error as E, state, value, Expr, Expression, Object, Result, TypeDef,
+    Value,
+};
+
+#[derive(thiserror::Error, Clone, Debug, PartialEq)]
+pub enum Error {
+    #[error(r#"invalid value type (got "{0}")"#)]
+    Invalid(value::Kind),
+}
+
+#[derive(Clone)]
+pub struct Argument {
```

Comment:
I find it confusing that we have multiple objects called `Argument` now.  We have `expression::Argument` and `function::Argument`.

Perhaps the `function::Argument` could be renamed `function::ArgumentType`?